### PR TITLE
stop results table from selecting all when a new fit is performed.

### LINKED ISSUE #32470: Results tab parameter tables are re-selected after a fit
**Original reporter:** James Lord

**Describe the bug**
The parameter tables in the results tab get re-selected whenever a fit is performed. This is not desired behaviour.

**To Reproduce**
1. Muon Analysis
2. Load MUSR 62260
3. Go Fitting tab and add a GausOsc function
4. Do a fit
5. Go to results tab and unselect the parameters table you just created
6. Go to fitting tab and step to a different group
7. Do a fit
8. Go to results tab and you can see it has re-selected the parameters table you had unselected

**Platform/Version (please complete the following information):**
Windows

### LINKED ISSUE #32726: Order of string labels in model plotting
In model analysis plots the users want strings to have a specific order (e.g. the time's ascending) . We need to check exactly which order for the string logs.

### PR DESCRIPTION
**Pull request by Harrietbrown**
**Description of work.**

This PR makes sure that the results tab retains the previously selected parameters when a new fit is performed.

**Report to:** James Lord

**To test:**

1. Muon Analysis
2. Load MUSR 62260
3. Go Fitting tab and add a GausOsc function
4. Do a fit
5. Go to results tab and unselect the parameters table you just created
6. Go to fitting tab and step to a different group
7. Do a fit
8. Go to results tab and you can see that the parameters to deselected are still unselected

Fixes #32470

---

#### Reviewer ####

Please comment on the following ([full description](http://developer.mantidproject.org/ReviewingAPullRequest.html)):

##### Code Review #####

- Is the code of an acceptable quality?
- Does the code conform to the [coding standards](http://developer.mantidproject.org/Standards/)?
- Are the unit tests small and test the class in isolation?
- If there is GUI work does it follow the [GUI standards](http://developer.mantidproject.org/Standards/GUIStandards.html)?
- If there are changes in the release notes then do they describe the changes appropriately?

##### Functional Tests #####

- Do changes function as described? Add comments below that describe the tests performed?
- Do the changes handle unexpected situations, e.g. bad input?
- Has the relevant (user and developer) documentation been added/updated?

Does everything look good? Mark the review as **Approve**. A member of `@mantidproject/gatekeepers` will take care of it.

### CODE REVIEW

File: qt/python/mantidqtinterfaces/mantidqtinterfaces/Muon/GUI/Common/results_tab_widget/results_tab_presenter.py [hunk 1]
```diff
@@ -107,6 +107,11 @@ def _update_fit_results_view_on_new_fit(self):
         """Update the view of results workspaces based on the current model"""
         workspace_list, function_name = self._get_workspace_list()
         self.view.set_selected_fit_function(function_name)
+
+        current_view = self.view.fit_result_workspaces()
+        not_selected = [key for key in current_view.keys() if not current_view[key][1]]
+        workspace_list = [ws for ws in workspace_list if ws not in not_selected]
```
Comment (AnthonyLim23) [inline qt/python/mantidqtinterfaces/mantidqtinterfaces/Muon/GUI/Common/results_tab_widget/results_tab_presenter.py:113]: could we not do something like the below? Or is there a reason this is not possible?
```
for ws in reversed(workspace_list):
     if not current_view[ws][1]:
              workspace_list.remove(ws)

```

Comment (Harrietbrown) [reply qt/python/mantidqtinterfaces/mantidqtinterfaces/Muon/GUI/Common/results_tab_widget/results_tab_presenter.py:113]: The current view does not have the new ws in it, so we can not rely on current_view[ws] nt failing.
We could add an `if ws in current_view.keys():` but I'm not sure if there is any value in that.

Comment (AnthonyLim23) [reply qt/python/mantidqtinterfaces/mantidqtinterfaces/Muon/GUI/Common/results_tab_widget/results_tab_presenter.py:113]: ok, just asking as it seemed odd to me to make a list, to then make another list based of a 3rd list. If thats the best way to do it, there is not much else we can do.

File: docs/source/release/v6.3.0/muon.rst [hunk 2]
```diff
@@ -98,5 +98,6 @@ Improvements
   - A bug has been fixed that caused Model fitting to not update it's results table list.
   - Plotting in Model fitting now features a greater number of units for parameters and sample logs.
   - The dates and times for relevant parameters in model fitting have been formatted so that they can be plotted with relative spacing.
+  - When a new fit is performed in Muon Analysis it no longer reselects all parameter workspaces in the results tab.
```
Comment (AnthonyLim23) [inline docs/source/release/v6.3.0/muon.rst:101]: lets put this in the muon +frequency analysis section

**Review by AnthonyLim23 — APPROVED** [review]
Just tested this and it works as expected and is a good quality of life improvement. :shipit:

**Pull request merged**